packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -492,6 +504,12 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
       this.node.addWarning('desiredCapacity has been configured. Be aware this will reset the size of your AutoScalingGroup on every deployment. See https://github.com/aws/aws-cdk/issues/5215');
     }
 
+    this.maxInstanceLifetime = props.maxInstanceLifetime ? props.maxInstanceLifetime : undefined;
+    if (this.maxInstanceLifetime &&
```

Comment:
To satisfy the previous comment you can preform the validation on the props value. 

From the docs it seems the 0 is also an allowed value and even have a specific meaning:

> When configuring the maximum instance lifetime for your Auto Scaling group, you must specify a value of at least 604,800 seconds (7 days). To clear a previously set value, specify a new value of 0.

Is 0 the equivalent of not setting a value in the template? If a value is first set and then removed is it the equivalent of setting 0?

Reply:
Agree. Will allow `0` in here. Thanks.

Reply:
Before we add it, do you know if setting the 0 value is required when creating the auto scaling group via CloudFormation? or is it only required when using other AWS tools such as the console/AWS CLI?

Reply:
@NetaNir Let me test it and note here. Thanks for bringing this up.

Reply:
Hi @NetaNir 

If we change the `maxInstanceLifetime`  from an arbitrary number to `0`, it will just reset its value to `undefined` from the console's perspective.

```ts
new autoscaling.AutoScalingGroup(stack, 'Fleet', {
  vpc,
  instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.MICRO),
  machineImage: new ec2.AmazonLinuxImage({ generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2 }),
  maxInstanceLifetime: cdk.Duration.days(0),
});
```
cdk diff the stack

<img width="1137" alt="圖片" src="https://user-images.githubusercontent.com/278432/81634232-230c1180-9441-11ea-8547-4eab324c648b.png">

And the value is empty from console as well.
<img width="647" alt="圖片" src="https://user-images.githubusercontent.com/278432/81634292-3fa84980-9441-11ea-9bdb-134c5b4fda90.png">

Setting to `0` seems be the same effect with simply leaving this property undefined.

Reply:
@NetaNir I've updated the description in the README as well as the docstring.

Reply:
Oh wow! thank you so much for figuring this out.
So given that, seems like we should lose the 0 value? Given that we can achieve the same result by simply removing the property I think it will be a cleaner API.
What do you think?

Reply:
Agree. Let me remove it and commit again.

Reply:
Thanks for all of your contributions to the cdk @pahud!

Reply:
No problem. Please review it again. Thanks.